src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/StringBuilderAppendShouldNotTakeSubstring.Fixer.cs

```diff
@@ -0,0 +1,191 @@
+            var lengthArgument = generator
+                .SubtractExpression(
+                    lengthNode,
+                    startIndexNode);
+
+            // generate sb.Append(text, 2, text.Length - 2)
+            var newNode = generator.InvocationExpression(
+                fixComponents.TargetMethod, 
+                fixComponents.StringArgument.Syntax,
+                startIndexNode, 
+                lengthArgument);
+
+            // replace sb.Append(text.Substring(2)) by sb.Append(text, 2, text.Length - 2)
+            var newRoot = root
+                .ReplaceNode(
+                    nodeToFix, 
+                    newNode);
+            return document.WithSyntaxRoot(newRoot);
+        }
+
+        private struct FixComponents
+        {
+            public FixComponents(
+                IOperation stringArgument, 
+                SyntaxNode targetMethod, 
+                ImmutableArray<IArgumentOperation> originalInnerArguments)
+            {
+                StringArgument = stringArgument;
+                TargetMethod = targetMethod;
+                OriginalInnerArguments = originalInnerArguments;
+            }
+
+            public IOperation StringArgument { get; }
+            public SyntaxNode TargetMethod { get; }
+            public ImmutableArray<IArgumentOperation> OriginalInnerArguments { get; }
+        }
+
+        private static async Task<Document> FixCodeTwoParameters(
+            Document document,
+            TextSpan span,
+            CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var nodeToFix = root.FindNode(span, getInnermostNodeForTie: true);
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            if (semanticModel.GetOperation(nodeToFix) is IInvocationOperation typedNodeToFix)
+            {
+                var generator = SyntaxGenerator.GetGenerator(document);
+
+                var fixComponents = GetFixComponents(generator, typedNodeToFix);
+
+                var (startIndexArgument, lengthArgument) = fixComponents.OriginalInnerArguments[0].Parameter.Name == "length"
+                    ? (fixComponents.OriginalInnerArguments[1], fixComponents.OriginalInnerArguments[0])
+                    : (fixComponents.OriginalInnerArguments[0], fixComponents.OriginalInnerArguments[1]);
+                
+                // from sb.Append(text.Substring(2, 5)) generate sb.Append(text, 2, 5)
+                var newNode = generator.InvocationExpression(
+                    fixComponents.TargetMethod,
+                    fixComponents.StringArgument.Syntax,
+                    startIndexArgument.Value.Syntax,
+                    lengthArgument.Value.Syntax);
+                var newRoot = root.ReplaceNode(
+                    nodeToFix,
+                    newNode);
```

Comment:
we really don't need to spread simple method calls over 3 lines.  it really just reduce the amount of code that can be effectively seen and absorbed while reading.  I thin kit's fine for the tree construction, or when there's a significant enogh code that horizontally there would be a problem.  but taking something like `foo.bar(a, b, c, d)` and making that 5 lines is very excessive.